controllers/topics-controller.js

```diff
@@ -0,0 +1,50 @@
+const { selectTopics, readTopics, readTopicsBySlug } = require('../models/topics-model');
+
+const getAllTopics = (request, response, next) => {
+    selectTopics().then((topics) => {
+        response.status(200).send(topics);  
+    })
+    .catch((err) => {
+        next(err);
+    });
+};
+
+const getTopicBySlug = (request, response, next) => {
+    const {slug} = request.params;
+    if(!isNaN(slug)){
+        throw {status: 400};
```

Comment:
You are not throwing an error here. But, for what it is worth, a throw is not as good as a Promise.reject here for two reason. 1. it only allows us to pass a string so actually this object would be totally ignored and the status, error etc will all come from the Error thrown by the server or Express, 2.  throwing inside a Promise triggers a reject but it won't be passed to the catch in our promise chain and then to app; the error will just throw immediately so we can't customise what it does